openpgp/packet/private_key.go

```diff
@@ -404,7 +404,18 @@ func (pk *PrivateKey) Decrypt(passphrase []byte) error {
 		data = data[:len(data)-2]
 	}
 
-	return pk.parsePrivateKey(data)
+	err := pk.parsePrivateKey(data)
+	if err != nil {
+		return err
+	}
+
+	if !pk.Dummy() {
```

Comment:
I think `pk.Dummy()` isn't possible at this point (see the beginning of the function)